libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -741,17 +740,37 @@ void CompilerUtils::convertType(
 		else if (targetTypeCategory == Type::Category::FixedPoint)
 		{
 			solAssert(
-				stackTypeCategory == Type::Category::Integer || 
+				stackTypeCategory == Type::Category::Integer ||
 				stackTypeCategory == Type::Category::RationalNumber ||
 				stackTypeCategory == Type::Category::FixedPoint,
 				"Invalid conversion to FixedMxNType requested."
 			);
-			//shift all integer bits onto the left side of the fixed type
+
 			FixedPointType const& targetFixedPointType = dynamic_cast<FixedPointType const&>(_targetType);
-			if (auto typeOnStack = dynamic_cast<IntegerType const*>(&_typeOnStack))
-				if (targetFixedPointType.numBits() > typeOnStack->numBits())
-					cleanHigherOrderBits(*typeOnStack);
-			solUnimplemented("Not yet implemented - FixedPointType.");
+			if (stackTypeCategory == Type::Category::Integer)
+			{
+				IntegerType const& typeOnStack = dynamic_cast<IntegerType const&>(_typeOnStack);
+				cleanHigherOrderBits(typeOnStack);
+				convertFixedPointType(0, targetFixedPointType.fractionalDigits(), false); // Signed doesn't matter, only multiplying
+				if (_cleanupNeeded)
+					cleanHigherOrderBits(targetFixedPointType);
+			}
+			else if (stackTypeCategory == Type::Category::RationalNumber)
+			{
+				auto typeOnStack = dynamic_cast<RationalNumberType const*>(&_typeOnStack);
+				solAssert(typeOnStack, "");
+				convertType(*typeOnStack->fixedPointType(), targetFixedPointType);
+			}
+			else
+			{
+				solAssert(stackTypeCategory == Type::Category::FixedPoint, "");
+				auto typeOnStack = dynamic_cast<FixedPointType const*>(&_typeOnStack);
+				solAssert(typeOnStack, "");
+				cleanHigherOrderBits(*typeOnStack);
+				convertFixedPointType(typeOnStack->fractionalDigits(), targetFixedPointType.fractionalDigits(), typeOnStack->isSigned());
```

Comment:
Should this case have an assertion for `targetFIxedPoinType.numBits() >= typeOnStack->numBits()` ?

Reply:
No, that is not a requirement. The helper function handles that.